### DISCUSSION TITLE
Use `ruff` instead of `flake8` + `isort`

### DISCUSSION THREAD

**Discussion by Kludex** [encode/httpx#2607] [Ideas]
It looks like _everybody_ is using it: https://ruff.pages.dev/

Should we use it as well on encode projects? cc @tomchristie @florimondmanca @adriangb

**Comment by adriangb** [comment]
I would be open to it, yes

**Comment by florimondmanca** [comment]
Oh, looking neat. Is the speedup that significant and worthwhile? I’ll try on a couple of libraries of mine. If this doesn’t cause too much breakage I’d also be open to it.

**Comment by Kludex** [comment]
I've created https://github.com/encode/httpx/pull/2648 cc @tomchristie
